apps/prairielearn/src/pages/instructorAssessmentQuestions/components/InstructorAssessmentQuestionsTable.tsx

```diff
@@ -47,81 +58,915 @@ function Title({
   return title;
 }
 
+function EditModeButtons({
+  csrfToken,
+  origHash,
+  zones,
+  editMode,
+  setEditMode,
+}: {
+  csrfToken: string;
+  origHash: string;
+  zones: ZoneAssessmentJson[];
+  editMode: boolean;
+  setEditMode: (editMode: boolean) => void;
+}) {
+  if (!editMode) {
+    return (
+      <button class="btn btn-sm btn-light" type="button" onClick={() => setEditMode(true)}>
+        <i class="fa fa-edit" aria-hidden="true" /> Edit questions
+      </button>
+    );
+  }
+
+  return (
+    <form method="POST">
+      <input type="hidden" name="__action" value="save_questions" />
+      <input type="hidden" name="__csrf_token" value={csrfToken} />
+      <input type="hidden" name="orig_hash" value={origHash} />
+      <input type="hidden" name="zones" value={JSON.stringify(zones)} />
+      <span class="js-edit-mode-buttons">
+        <button class="btn btn-sm btn-light mx-1" type="submit">
+          <i class="fa fa-save" aria-hidden="true" /> Save and sync
+        </button>
+        <button class="btn btn-sm btn-light" type="button" onClick={() => window.location.reload()}>
+          Cancel
+        </button>
+      </span>
+    </form>
+  );
+}
+
+function questionDisplayName(course: StaffCourse, question: StaffAssessmentQuestionRow) {
+  if (idsEqual(course.id, question.question.course_id)) {
```

Comment:
Nit: it shouldn't be necessary to use `idsEqual` here since both `course` and `question` should have string IDs. This is just left over from a past where IDs were sometimes numbers and sometimes strings.